checker/src/main/java/org/checkerframework/checker/calledmethods/framework/FrameworkSupportUtils.java

```diff
@@ -0,0 +1,76 @@
+package org.checkerframework.checker.calledmethods.framework;
+
+import java.util.EnumSet;
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.calledmethods.CalledMethodsChecker;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/** A utility class of static methods used in supporting builder-generation frameworks. */
+public class FrameworkSupportUtils {
+    /** This class is non-instantiable */
+    private FrameworkSupportUtils() {}
+
+    /**
+     * Checks whether the given type is one of the immutable collections defined in
+     * com.google.common.collect.
+     *
+     * @param type a Java type
+     * @return whether the type is a Guava immutable collection
+     */
+    public static boolean isGuavaImmutableType(TypeMirror type) {
+        return type.toString().startsWith("com.google.common.collect.Immutable");
+    }
+
+    /**
+     * Capitalizes the first letter of the given string
+     *
+     * @param prop a String
+     * @return the same String, but with the first character capitalized
+     */
+    public static String capitalize(String prop) {
+        return prop.substring(0, 1).toUpperCase() + prop.substring(1);
+    }
+
+    /**
+     * Given an annotation name, return true if the element has the annotation of that name
+     *
+     * @param element the element
+     * @param annotName name of the annotation
+     * @return true if the element has the annotation of that name
+     */
+    public static boolean hasAnnotation(Element element, String annotName) {
```

Comment:
I suggest putting this in `ElementUtils`.

Reply:
The `withValues` bit is an example - it could have been any method in `Filter`. The comment is a bit confusing, so I will rewrite it to be more clear.